x-pack/metricbeat/module/aws/ec2/ec2.go

```diff
@@ -0,0 +1,330 @@
+	os.Setenv("AWS_SESSION_TOKEN", config.SessionToken)
+
+	metricSet, err := aws.NewMetricSet(base)
+	if err != nil {
+		return nil, errors.Wrap(err, "error creating aws metricset")
+	}
+
+	return &MetricSet{
+		MetricSet: metricSet,
+		config:    &config,
+	}, nil
+}
+
+// Fetch methods implements the data gathering and data conversion to the right
+// format. It publishes the event which is then forwarded to the output. In case
+// of an error set the Error field of mb.Event or simply call report.Error().
+func (m *MetricSet) Fetch(report mb.ReporterV2) {
+	//mock Fetch
+	if m.config.Mock == "true" {
+		m.MockFetch(report)
+		return
+	}
+
+	//actual fetch function
+	cfg, err := external.LoadDefaultAWSConfig()
+	if err != nil {
+		fmt.Println("Failed to load config: ", err.Error())
+	}
+
+	cfg.Region = "us-west-1"
+	svcEC2 := ec2.New(cfg)
+	//Get a list of regions
+	regionsList, err := getRegions(svcEC2)
+	if err != nil {
+		report.Error(errors.Wrap(err, "getRegions failed"))
+	}
+
+	for _, regionName := range regionsList {
+		cfg.Region = regionName
+		svcEC2 := ec2.New(cfg)
+		instanceIDs, instancesOutputs, err := getInstancesPerRegion(svcEC2)
+		if err != nil {
+			report.Error(errors.Wrap(err, "getInstancesPerRegion failed"))
+		}
+
+		svcCloudwatch := cloudwatch.New(cfg)
+		for _, instanceID := range instanceIDs {
+			init := true
+			getMetricDataOutput := &cloudwatch.GetMetricDataOutput{NextToken: nil}
+			for init || getMetricDataOutput.NextToken != nil {
+				init = false
+				output, err := getMetricDataPerRegion(instanceID, getMetricDataOutput.NextToken, svcCloudwatch)
+				if err != nil {
+					report.Error(errors.Wrap(err, "getMetricDataPerRegion failed"))
+				}
+				getMetricDataOutput.MetricDataResults = append(getMetricDataOutput.MetricDataResults, output.MetricDataResults...)
+			}
+			reportCloudWatchEvents(getMetricDataOutput, instanceID, instancesOutputs[instanceID], regionName, report)
+		}
+	}
+}
+
+// MockFetch methods implements the data gathering and data conversion using MockEC2Client and MockCloudWatchClient
+func (m *MetricSet) MockFetch(report mb.ReporterV2) {
+	svcEC2Mock := &MockEC2Client{}
+	instanceIDs, instancesOutputs, err := getInstancesPerRegion(svcEC2Mock)
+	if err != nil {
+		report.Error(errors.Wrap(err, "getInstancesPerRegion failed"))
+	}
+
+	svcCloudwatchMock := &MockCloudWatchClient{}
+	for _, instanceID := range instanceIDs {
+		getMetricDataOutput, err := getMetricDataPerRegion(instanceID, nil, svcCloudwatchMock)
+		if err != nil {
+			report.Error(errors.Wrap(err, "getMetricDataPerRegion failed"))
+		}
+		reportCloudWatchEvents(getMetricDataOutput, instanceID, instancesOutputs[instanceID], "us-west-1", report)
+	}
+}
+
+func getRegions(svc ec2iface.EC2API) (regionsList []string, err error) {
+	input := &ec2.DescribeRegionsInput{}
+	req := svc.DescribeRegionsRequest(input)
+	output, err := req.Send()
+	if err != nil {
+		fmt.Println("Failed DescribeRegions: ", err)
```

Comment:
leftover?

Reply:
I see there are several of these, you should use `logp.Err` where you want to show the error in the log, or just remove the Print statement if you want to pass it